modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdPInitiatedSSOTestCase.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.identity.integration.test.util.Utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.integration.test.utils.CommonConstants.IS_DEFAULT_HTTPS_PORT;
+
+/**
+ * This class contains the test method for IDP initiated SAML SSO login process with the relevant other behaviors.
+ */
+public class SAMLIdPInitiatedSSOTestCase extends AbstractSAMLSSOTestCase {
+
+    private Log log = LogFactory.getLog(getClass());
+
+    private static final String SP_NAME = "travelocity.com";
+    private static final String IDP_INIT_SSO_URL = "https://localhost:%s/samlsso?spEntityID=travelocity.com";
+    private static final String SAML_SSO_URL = "https://localhost:%s/samlsso";
+    private static final String SP_ACS_URL = "http://localhost:8490/%s/home.jsp";
+    private HttpClient httpClient;
+    private CookieStore cookieStore = new BasicCookieStore();
+    private AbstractSAMLSSOTestCase.SAMLConfig samlConfig;
+
+    @Factory(dataProvider = "samlConfigProvider")
+    public SAMLIdPInitiatedSSOTestCase(SAMLConfig config) {
+
+        if (log.isDebugEnabled()) {
+            log.info("SAML SSO Test initialized for " + config);
+        }
+        this.samlConfig = config;
+    }
+
+    @DataProvider(name = "samlConfigProvider")
+    public static Object[][] samlConfigProvider() {
+
+        return new SAMLConfig[][]{
+                {new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, User.SUPER_TENANT_USER, HttpBinding.HTTP_REDIRECT,
+                        ClaimType.NONE, App.SUPER_TENANT_APP_WITH_SIGNING)},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initTest() throws Exception {
+
+        super.init();
+        super.testInit();
+        addServiceProvider();
+        httpClient = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).disableRedirectHandling().build();
+        super.createUser(samlConfig);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        super.deleteUser(samlConfig);
+        super.ssoConfigServiceClient.removeServiceProvider(SP_NAME);
+        super.deleteApplication(SP_NAME);
+        super.testClear();
+    }
+
+    @Test(groups = "wso2.is", description = "This test method will test the IdP initiated SAML SSO process.")
+    public void testIdPInitiatedSSO() throws Exception {
```

Comment:
Do we have any specific reason to throw Exception?

Reply:
From the upper level codes, exception class is thrown. So need to keep this as it is.